internal/provider/data_source_azure_peering_connection.go

```diff
@@ -0,0 +1,147 @@
+package provider
+
+import (
+	"context"
+	"log"
+
+	networkmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-network/preview/2020-09-07/models"
+	sharedmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-shared/v1/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+)
+
+func dataSourceAzurePeeringConnection() *schema.Resource {
+	return &schema.Resource{
+		Description: "The Azure peering connection data source provides information about a peering connection between an HVN and a peer Azure VNet.",
+		ReadContext: dataSourceAzurePeeringConnectionRead,
+		Timeouts: &schema.ResourceTimeout{
+			Default: &peeringCreateTimeout,
+		},
+		Schema: map[string]*schema.Schema{
+			// Required inputs
+			"peering_id": {
+				Description: "The ID of the peering connection.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"hvn_link": {
+				Description: "The `self_link` of the HashiCorp Virtual Network (HVN).",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			// Optional inputs
+			"wait_for_active_state": {
```

Comment:
If `hcp_peering_connection_activation` is used to wait for peering connection to reach ACTIVE then is this field required?

Reply:
Actually the docs are outdated. The activation datasource was the original solution, but we decided we'd rather follow an existing TGW pattern and use this `wait_for_active_state` flag on the peering datasource.